src/modules/hip/kernel/color_twist.hpp

```diff
@@ -0,0 +1,339 @@
+    *pixelB = fmaf(bf, colorTwistParams_f4->x, colorTwistParams_f4->y);
+}
+
+__device__ void color_twist_8RGB_hip_compute(d_float24 *pix_f24, float4 *colorTwistParams_f4)
+{
+    color_twist_1RGB_hip_compute(&(pix_f24->x.x.x), &(pix_f24->y.x.x), &(pix_f24->z.x.x), colorTwistParams_f4);
+    color_twist_1RGB_hip_compute(&(pix_f24->x.x.y), &(pix_f24->y.x.y), &(pix_f24->z.x.y), colorTwistParams_f4);
+    color_twist_1RGB_hip_compute(&(pix_f24->x.x.z), &(pix_f24->y.x.z), &(pix_f24->z.x.z), colorTwistParams_f4);
+    color_twist_1RGB_hip_compute(&(pix_f24->x.x.w), &(pix_f24->y.x.w), &(pix_f24->z.x.w), colorTwistParams_f4);
+    color_twist_1RGB_hip_compute(&(pix_f24->x.y.x), &(pix_f24->y.y.x), &(pix_f24->z.y.x), colorTwistParams_f4);
+    color_twist_1RGB_hip_compute(&(pix_f24->x.y.y), &(pix_f24->y.y.y), &(pix_f24->z.y.y), colorTwistParams_f4);
+    color_twist_1RGB_hip_compute(&(pix_f24->x.y.z), &(pix_f24->y.y.z), &(pix_f24->z.y.z), colorTwistParams_f4);
+    color_twist_1RGB_hip_compute(&(pix_f24->x.y.w), &(pix_f24->y.y.w), &(pix_f24->z.y.w), colorTwistParams_f4);
+}
+
+__device__ void color_twist_hip_compute(uchar *srcPtr, d_float24 *pix_f24, float4 *colorTwistParams_f4)
+{
+    float4 normalizer_f4 = (float4) ONE_OVER_255;
+    rpp_hip_math_multiply24_const(pix_f24, pix_f24, normalizer_f4);
+    colorTwistParams_f4->x = colorTwistParams_f4->x * 255.0f;
+    colorTwistParams_f4->z = (((int)colorTwistParams_f4->z) % 360) * SIX_OVER_360;
+    color_twist_8RGB_hip_compute(pix_f24, colorTwistParams_f4);
+    rpp_hip_pixel_check_0to255(pix_f24);
+}
+__device__ void color_twist_hip_compute(float *srcPtr, d_float24 *pix_f24, float4 *colorTwistParams_f4)
+{
+    colorTwistParams_f4->y = colorTwistParams_f4->y * ONE_OVER_255;
+    colorTwistParams_f4->z = (((int)colorTwistParams_f4->z) % 360) * SIX_OVER_360;
+    color_twist_8RGB_hip_compute(pix_f24, colorTwistParams_f4);
+    rpp_hip_pixel_check_0to1(pix_f24);
+}
+__device__ void color_twist_hip_compute(half *srcPtr, d_float24 *pix_f24, float4 *colorTwistParams_f4)
+{
+    colorTwistParams_f4->y = colorTwistParams_f4->y * ONE_OVER_255;
+    colorTwistParams_f4->z = (((int)colorTwistParams_f4->z) % 360) * SIX_OVER_360;
+    color_twist_8RGB_hip_compute(pix_f24, colorTwistParams_f4);
+    rpp_hip_pixel_check_0to1(pix_f24);
+}
+__device__ void color_twist_hip_compute(schar *srcPtr, d_float24 *pix_f24, float4 *colorTwistParams_f4)
+{
+    float4 i8Offset_f4 = (float4) 128.0f;
+    float4 normalizer_f4 = (float4) ONE_OVER_255;
+    rpp_hip_math_add24_const(pix_f24, pix_f24, i8Offset_f4);
+    rpp_hip_math_multiply24_const(pix_f24, pix_f24, normalizer_f4);
+    colorTwistParams_f4->x = colorTwistParams_f4->x * 255.0f;
+    colorTwistParams_f4->z = (((int)colorTwistParams_f4->z) % 360) * SIX_OVER_360;
+    color_twist_8RGB_hip_compute(pix_f24, colorTwistParams_f4);
+    rpp_hip_pixel_check_0to255(pix_f24);
+    rpp_hip_math_subtract24_const(pix_f24, pix_f24, i8Offset_f4);
```

Comment:
what is i8Offset_f4 used for?

Reply:
Its just for the i8 range offset to -128 to +127.